admin-console/src/main/java/com/agiletec/apsadmin/portal/helper/AbstractPageActionHelper.java

```diff
@@ -101,11 +103,14 @@ public ITreeNode getAllowedTreeRoot(Collection<String> userGroupCodes, boolean a
 	}
 
 	private void addTreeWrapper(PageTreeNodeWrapper currentNode, IPage currentTreeNode, Collection<String> userGroupCodes, boolean alsoFreeViewPages) {
-		IPage[] children = currentTreeNode.getChildren();
+		String[] children = currentTreeNode.getChildrenCodes();
 		for (int i = 0; i < children.length; i++) {
-			IPage newCurrentTreeNode = children[i];
+			IPage newCurrentTreeNode = (this.isDraftPageHepler())
+					? this.getPageManager().getDraftPage(children[i])
+					: this.getPageManager().getOnlinePage(children[i]);

```

Comment:
See above. Move to a utility class.

Reply:
See the reply for the class PageAction.java